draft-ietf-ppm-dap.md

```diff
@@ -1295,7 +1296,13 @@ An aggregation job can be thought of as having three phases:
 - Completion: Finish the aggregate flow, yielding an output share corresponding
   to each report share in the aggregation job.
 
-These phases are described in the following subsections.
+After an aggregation job is completed, each Aggregator stores the output shares
+until the aggregate share is collected as described in {{collect-flow}}. Note
+that it is usually not necessary to store output shares individually: depending
+on the batch mode and VDAF, the output shares can be merged into existing
+aggregate shares that are updated as aggregation jobs complete. This streaming
+aggregation is compatible with Prio3 and all batch modes specified in this
+document.
```

Comment:
I wonder if it would be clearer to phrase this the other way: aggregation is part of the completion phase of each aggregation job, _but_ some future VDAF might require "delayed aggregation" (but probably not).

Reply:
Personally, I think the likelihood of any VDAF using delayed aggregation as a feature is remote enough that we don't have too twist too hard to anticipate it.

Reply:
Right.  So I'm imagining that we say "Completion of an aggregation job includes aggregation" and then add some fine print about delayed aggregation.  That's the opposite of this proposal, which says "Completion actually only means VDAF preparation", and then notes that everyone will probably do streaming aggregation.

Reply:
I see what you're getting at. Let me give it a try.

Reply:
Argh, there's a problem. To specify this rigorously, we need two things:

1. The concept of a "batch bucket" (#385). The definition of "batch bucket" depends on the batch mode: for time-interval, it's a window of time; for leader-selected, it's simply a batch ID. Each bucket has an aggregate share. During the "completion phase" of an aggregation job, we would add each output share into the aggregate share of the bucket it corresponds to.

2. We need to be able to express syntactically how to update an aggregate share with an output share. This is precluded by the current VDAF syntax, [which only describes aggregating output shares into an aggregate share](https://cfrg.github.io/draft-irtf-cfrg-vdaf/draft-irtf-cfrg-vdaf.html#section-5.4). We would therefore need to modify the syntax.

We want (1.) in any case, however I think (2.) requires some discussion. For what it's worth I'd before modifying the syntax to account for this, and I even argued for doing this in the past, but at the time there was a strong preference to keep the abstraction boundary where it is.

I propose we merge this PR without this suggestion and kick off a discussion.

Reply:
Fine with me, but you might want to adjust the description so it doesn't close the Github issue.

Reply:
I think the discussion is broader than the text of the issue, so I'd prefer to close and open a new issue to discuss making per-bucket aggregation normative.